### DISCUSSION TITLE
[@types/braintree] bug in subscription.update request type

### DISCUSSION THREAD

**Discussion by pdeveltere** [DefinitelyTyped/DefinitelyTyped#60610] [Issues with a @types package] [answered]
There is a bug in the braintree types line 327: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/braintree/index.d.ts#L327

the function `braintree.subscription.update`, is typed the same as the `braintree.subscription.create` function. However, the options parameter is not the same. As seen in the documentation:
https://developer.paypal.com/braintree/docs/reference/request/subscription/update/node#options
https://developer.paypal.com/braintree/docs/reference/request/subscription/create/node#options

e.g. this doesn't work:
```
const response = await braintree.subscription.update(id, {
  options: { prorateCharges: false },
});
```

[a comment by an automated account is not shown]

**Comment by smrubin** [comment, marked as answer]
Thanks for surfacing this @pdeveltere. I think this could be as simple as splitting the `SubscriptionRequest` interface into a `SubscriptionCreateRequest` and `SubscriptionUpdateRequest` interface. This would follow similar patterns with the rest of the typescript definition, when the create and update request payloads are different.

@pdeveltere Any interest in submitting a PR for this?

**Reply by pdeveltere** [reply]
I'm create a PR 👍🏽

**Reply by pdeveltere** [reply]
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/60758
